utilities/dbviz/src/postgresql.rs

```diff
@@ -0,0 +1,312 @@
+//! Loader for postgresql.
+
+// This is not production code, so while we should get rid of these lints, its not
+// worth the time it would take, for no return value.
+#![allow(clippy::unwrap_used)]
+#![allow(clippy::indexing_slicing)]
+
+use std::{
+    cell::RefCell,
+    collections::HashMap,
+    convert::{TryFrom, TryInto},
+};
+
+use anyhow::Result;
+use itertools::Itertools;
+use postgres::{tls::NoTls, Client, Row};
+
+use crate::{
+    opts,
+    schema::{Index, Relation, Schema, Table, TableColumn},
+};
+
+/// Struct that manages the loading and implements `Loader` trait.
+pub struct Conn {
+    /// Postgres client
+    pg_client: RefCell<Client>,
+    /// Schema name
+    schema: String,
+    /// Options
+    opts: opts::Cli,
+}
+
+/// Check if a column is a primary key
+fn is_primary_key(table: &str, column: &str, indexes: &[Index]) -> bool {
+    indexes
+        .iter()
+        .any(|idx| idx.table == table && idx.fields.contains(&column.to_string()) && idx.primary)
+}
+
+impl Conn {
+    /// Make a new postgres connection
+    pub(crate) fn new(opts: &opts::Cli) -> Result<Conn> {
+        let pg_client = postgres::Config::new()
+            .user(&opts.pg_opts.username)
+            .password(&opts.pg_opts.password)
+            .dbname(&opts.pg_opts.database)
+            .host(&opts.pg_opts.hostname)
+            .connect(NoTls)?;
+
+        let pg_client = RefCell::new(pg_client);
+        let schema = opts.pg_opts.schema.clone();
+        Ok(Conn {
+            pg_client,
+            schema,
+            opts: opts.clone(),
+        })
+    }
+
+    /// Do we include this table name?
+    fn include_table(&self, name: &String) -> bool {
+        match &self.opts.include_tables {
+            Some(inc) => inc.contains(name),
+            None => true,
+        }
+    }
+
+    /// Do we include this table name?
```

Comment:
Exclude

Reply:
For things like this, it's better and more efficient to actually "suggest change", which can easily be accepted if the PR author agrees.  This saves a lot of time for both the reviewer and the author.